src/galileo/middleware/tracing.py

```diff
@@ -0,0 +1,204 @@
+"""
+Distributed tracing middleware for Starlette-based applications.
+
+This middleware automatically extracts distributed tracing headers from incoming HTTP requests
+and makes them available to the Galileo logger within request handlers.
+
+Works with any ASGI framework built on Starlette:
+- FastAPI
+- Starlette
+- Any other Starlette-based framework
+
+Example usage with FastAPI:
+    ```python
+    from fastapi import FastAPI
+    from galileo.middleware import TracingMiddleware, get_request_logger
+
+    app = FastAPI()
+    app.add_middleware(TracingMiddleware)
+
+    @app.post("/process")
+    async def process_request(data: dict):
+        # Logger automatically continues the distributed trace
+        logger = get_request_logger()
+        logger.add_workflow_span(input=str(data), name="process_workflow")
+        # ... process request ...
+        logger.conclude(output="done")
+        return {"status": "success"}
+    ```
+
+Example usage with Starlette:
+    ```python
+    from starlette.applications import Starlette
+    from starlette.routing import Route
+    from galileo.middleware import TracingMiddleware, get_request_logger
+
+    async def homepage(request):
+        logger = get_request_logger()
+        logger.add_workflow_span(input="homepage", name="homepage_handler")
+        logger.conclude(output="success")
+        return {"status": "ok"}
+
+    app = Starlette(
+        routes=[Route("/", homepage)],
+        middleware=[TracingMiddleware]
+    )
+    ```
+"""
+
+import logging
+
+from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint
```

Comment:
Suggestion: The file imports starlette at the top level. Since starlette is an optional dependency, simply importing galileo.middleware (e.g., by IDE auto-import or accidentally) could crash the app if the extra isn't installed. Consider wrapping the imports or deferring them.

Reply:
Approve with minor suggestions.

Optional Dependency Safety: Verify that importing galileo.middleware doesn't crash if starlette isn't installed (or document that the module shouldn't be imported without the extra).